lib/include/tick/array/sparsearray2d.h

```diff
@@ -106,6 +98,103 @@ class SparseArray2d : public BaseArray2d<T> {
   //! structure THUS the array becomes a view. \warning : This method cannot be
   //! called on a view
   std::shared_ptr<SSparseArray2d<T>> as_ssparsearray2d_ptr();
+
+  template <class Archive>
+  void save(Archive &ar) const {
+    try {
+      ar(this->_size_sparse);
+      ar(this->_n_rows);
+      ar(this->_n_cols);
+      ar(this->_size);
+
+      inner_save(ar);
+
+      std::vector<INDICE_TYPE> v_indices(this->_indices, this->_indices + this->_size_sparse),
+          v_row_indices(this->_row_indices, this->_row_indices + (this->_n_rows + 1));
+      ar(v_indices);
+      ar(v_row_indices);
+    } catch (const std::exception &e) {
+      std::cerr << e.what() << std::endl;
```

Comment:
same here

Reply:
an exception on saving is unlikely, but sometimes better safe than sorry

perhaps we may want a "Array.isValid()" or ".isLoaded()" that exists as a flag for this